sdk/samples/core/mosquitto_rpc_server_sample.c

```diff
@@ -368,17 +367,16 @@ int main(int argc, char* argv[])
 
   az_mqtt5_rpc_server_codec_options server_codec_options
       = az_mqtt5_rpc_server_codec_options_default();
-  server_codec_options.subscription_topic_format = subscription_topic_format;
 
   LOG_AND_EXIT_IF_FAILED(az_mqtt5_rpc_server_init(
       &rpc_server,
       &rpc_server_codec,
       &mqtt_connection,
       property_bag,
       AZ_SPAN_FROM_BUFFER(subscription_topic_buffer),
+      service_group_id,
```

Comment:
Same as above - I think the codec should keep this stored (by passing it as an option). The RPC server should simply use the pre-configured codec.